setup.py

```diff
@@ -22,6 +22,8 @@
     "tiktoken",
     "pydantic>=1.10,<3",  # could be both V1 and V2
     "docker",
+    "jupyter-client>=8.6.0",
+    "ipykernel>=6.29.0",
```

Comment:
Do we have to add them here? I think these are heavier dependencies and shouldn't add them unless necessary. For example, if someone doesn't need code execution at all, they shouldn't be forced to install them. I learnt about this in a hard way when working on FLAML.

Reply:
We can include them as optional dependencies.

Reply:
Made optional. Skip tests if dependencies not installed.